src/controllers/LoginController.ts

```diff
@@ -16,31 +18,63 @@ export default class LoginController {
    * @returns {Promise<Response>} Promise that resolves to an Express response
    */
   static async login(req: Request, res: Response): Promise<Response> {
-    // Validate request body against login schema
-    const { error } = loginSchema.validate(req.body)
-    if (error) {
-      return res.status(400).json({ message: error.details[0].message })
-    }
+    try {
+      const { email, password } = req.body
 
-    const { email, password } = req.body
+      const existingUser = await User.findOne({ where: { email } })
 
-    const existingUser = await User.findOne({ where: { email } })
-    if (!existingUser) {
-      return res.status(401).json({ message: 'Invalid email or password' })
-    }
+      if (!existingUser) {
+        return res.status(404).send({ message: 'User not found' })
+      }
 
-    const isPasswordValid = await bcrypt.compare(
-      password,
-      existingUser.password,
-    )
-    if (!isPasswordValid) {
-      return res.status(401).json({ message: 'Invalid email or password' })
-    }
+      const passwordMatches = await comparePassword(
+        password,
+        existingUser.dataValues.password,
+      )
+      if (!passwordMatches) {
+        return res.status(401).send({ message: 'Invalid credentials' })
+      }
+
+      if (existingUser.dataValues.verified === false) {
+        // Handle pending verification
+        const pendingVerification = await Token.findOne({
+          where: { userId: existingUser.id },
+        })
 
-    const token = generateToken({
-      id: existingUser.id,
-      email: existingUser.email,
-    })
-    return res.status(200).json({ jwt: token, message: 'Login successful' })
+        if (pendingVerification) {
+          return res.status(200).send({ message: 'Please verify your email' })
+        }
+
+        // Generate and send new verification email
+        const newVerificationToken = generateToken({
+          id: existingUser.id,
+          email: existingUser.email,
+        })
+        const baseUrl = `${process.env.NODEMAILER_BASE_URL}users/verify/${newVerificationToken}`
+        const html = verificationsEmailTemplate(existingUser.username, baseUrl)
+
+        await Promise.all([
+          sendMail(email, 'Verify Your Account', html),
```

Comment:
By the way , this is good idea to resend a verification link to the user tried to login while they have not verified their account. Good job.